com.unity.render-pipelines.high-definition/Runtime/Lighting/AtmosphericScattering/Fog.cs

```diff
@@ -51,28 +47,50 @@ public class Fog : VolumeComponent
+        /// <summary>Volumetric fog anisotropy.</summary>
+        public ClampedFloatParameter anisotropy = new ClampedFloatParameter(0.0f, -1.0f, 1.0f);
         /// <summary>Controls the distribution of slices along the Camera's focal axis. 0 is exponential distribution and 1 is linear distribution.</summary>
         [Tooltip("Controls the distribution of slices along the Camera's focal axis. 0 is exponential distribution and 1 is linear distribution.")]
         public ClampedFloatParameter sliceDistributionUniformity = new ClampedFloatParameter(0.75f, 0, 1);
+
+        // Limit parameters for the fog quality
+        internal const float minFogScreenResolutionPercentage = (1.0f / 16.0f) * 100;
+        internal const float optimalFogScreenResolutionPercentage = (1.0f / 8.0f) * 100;
+        internal const float maxFogScreenResolutionPercentage = 0.5f * 100;
+        internal const int maxFogSliceCount = 512;
+
+        /// <summary>Two modes are available for controlling the performance and quality for the volumetric fog. The balance mode allows for an intuitive, performance oriented approach. The manual mode gives access to the internal set of parameters required for the evaluation.</summary>
+        [Tooltip("Two modes are available for controlling the performance and quality for the volumetric fog. The balance mode allows for an intuitive, performance oriented approach. The manual mode gives access to the internal set of parameters required for the evaluation.")]
+        public FogControlParameter fogControlMode = new FogControlParameter(FogControl.Balance);
         /// <summary>Resolution of the volumetric buffer (3D texture) along the X and Y axes relative to the resolution of the frame buffer.</summary>
         [Tooltip("Resolution of the volumetric buffer, along the x-axis and y-axis, relative to the resolution of the frame buffer.")]
-        public ClampedFloatParameter screenResolutionPercentage = new ClampedFloatParameter((1.0f/8.0f) * 100, (1.0f/16.0f) * 100, 100);
+        public ClampedFloatParameter screenResolutionPercentage = new ClampedFloatParameter(optimalFogScreenResolutionPercentage, minFogScreenResolutionPercentage, maxFogScreenResolutionPercentage);
         /// <summary>Number of slices of the volumetric buffer (3D texture) along the camera's focal axis.</summary>
         [Tooltip("Number of slices of the volumetric buffer (3D texture) along the camera's focal axis.")]
-        public ClampedIntParameter volumeSliceCount = new ClampedIntParameter(64, 1, 1024);
+        public ClampedIntParameter volumeSliceCount = new ClampedIntParameter(64, 1, maxFogSliceCount);
+        /// <summary>The slider controls in a linear way the cost and quality of the volumetric fog. The value 0 being the least expensive and 1 the highest quality.</summary>
+        [Tooltip("The slider controls in a linear way the cost and quality of the volumetric fog. The value 0 being the least expensive and 1 the highest quality.")]
```

Comment:
Controls the performance to quality ratio of the volumetric fog. A value of 0 being the least resource-intensive and a value of 1 being the highest quality.